src/Model.hs

```diff
@@ -1,138 +1,251 @@
--- | Where we define the model. Pretty much every module has to load this,
--- because various definitions need to know about User -- it would be nice to
--- decouple these.
-
 {-# LANGUAGE GADTs #-}
 {-# LANGUAGE TypeFamilies #-}
-{-# LANGUAGE DeriveGeneric #-}
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
-{-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE TemplateHaskell #-}
 {-# LANGUAGE QuasiQuotes #-}
 {-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings #-}
 
-module Model where
+{-@ LIQUID "--compile-spec" @-}
+
+module Model 
```

Comment:
I recommend not to look at the diff of this file because it changed drastically after it was automatically generated.